controllers/entandoappv2_controller.go

```diff
@@ -92,8 +138,10 @@ func (r *EntandoAppV2Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 	//log := r.Log.WithName("Upgrade Controller")
 	return ctrl.NewControllerManagedBy(mgr).
 		// FIXME! add filter on create for EntandoAppV2 cr
-		For(&v1alpha1.EntandoAppV2{}).
-		WithEventFilter(predicate.GenerationChangedPredicate{}). //solo modifiche a spec
+		For(&v1alpha1.EntandoAppV2{}, utils.ForOptionCustom()).
```

Comment:
custom predicates can be used in For and in Watches methods, in this way it is possible to filter the events based on the type of resource, for example different behaviors from CR and deployments events can be used